arangod/RocksDBEngine/RocksDBEngine.h

```diff
@@ -459,7 +461,7 @@ class RocksDBEngine final : public StorageEngine {
 
   class RocksDBSnapshot final : public StorageSnapshot {
    public:
-    explicit RocksDBSnapshot(rocksdb::DB& db) noexcept : _snapshot(&db) {}
+    explicit RocksDBSnapshot(rocksdb::DB& db) : _snapshot(&db) {}
```

Comment:
Is our code ready to have exception from this ctor?

Reply:
I think so. 
The only way to create `RocksDBSnapshot` is by calling the method `RocksDBEngine::currentSnapshot()`.
All of its 3 callers have wrapped the call into a try...catch.

Previously, if the ctor of `RocksDBSnapshot` would throw, the program would only std::terminate. Now it can continue, and the callers can handle the exception (looks like the callers expect exceptions anyway, because they already had try...catch blocks).

Reply:
Good